src/TestExplorer.ts

```diff
@@ -0,0 +1,153 @@
+import { TestExplorerParams } from "./TestExplorerParams";
+
+class CustomSigner implements msRest.ServiceClientCredentials {
+  private token: string;
+  constructor(token: string) {
+    this.token = token;
+  }
+
+  async signRequest(webResource: msRest.WebResourceLike): Promise<msRest.WebResourceLike> {
+    webResource.headers.set("authorization", `bearer ${this.token}`);
+    return webResource;
+  }
+}
+
+class TestExplorer {
+  private notebooksTestRunnerApplicationId: string;
+  private notebooksTestRunnerClientId: string;
+  private notebooksTestRunnerClientSecret: string;
+  private notebooksAccountName: string;
+  private notebooksAccountKey: string;
+  private notebooksAccountSubscriptonId: string;
+  private notebooksAccountResourceGroup: string;
+
+  constructor() {
+    window.onload = () => {
+      this.initTestExplorer();
+    };
+    window.addEventListener("message", this.handleMessage.bind(this), false);
+  }
+
+  private parseUrlParams = (): void => {
+    window.location.search
+      .substr(1)
+      .split("&")
+      .forEach(item => {
+        const tmp = item.split("=");
+        const value = decodeURIComponent(tmp[1]);
+        switch (tmp[0]) {
+          case TestExplorerParams.notebooksTestRunnerApplicationId:
+            this.notebooksTestRunnerApplicationId = value;
+            break;
+          case TestExplorerParams.notebooksTestRunnerClientId:
+            this.notebooksTestRunnerClientId = value;
+            break;
+          case TestExplorerParams.notebooksTestRunnerClientSecret:
+            this.notebooksTestRunnerClientSecret = value;
+            break;
+          case TestExplorerParams.notebooksAccountName:
+            this.notebooksAccountName = value;
+            break;
+          case TestExplorerParams.notebooksAccountKey:
+            this.notebooksAccountKey = value;
+            break;
+          case TestExplorerParams.notebooksAccountSubscriptonId:
+            this.notebooksAccountSubscriptonId = value;
+            break;
+          case TestExplorerParams.notebooksAccountResourceGroup:
+            this.notebooksAccountResourceGroup = value;
+            break;
+        }
+      });
+  };
+
+  private handleMessage(event: MessageEvent) {
+    if (event.data.type === MessageTypes.InitTestExplorer || event.data.type === MessageTypes.HideConnectScreen) {
+      this.sendMessageToExplorerFrame(event.data);
+    }
+  }
+
+  private async AADLogin(): Promise<string> {
+    const credentials = new ClientSecretCredential(
+      this.notebooksTestRunnerApplicationId,
+      this.notebooksTestRunnerClientId,
+      this.notebooksTestRunnerClientSecret
+    );
+    const token = await credentials.getToken("https://management.core.windows.net/.default");
+    return token.token;
+  }
+
+  private async getDatabaseAccount(token: string): Promise<DatabaseAccountsGetResponse> {
+    const client = new CosmosDBManagementClient(new CustomSigner(token), this.notebooksAccountSubscriptonId);
```

Comment:
Ugh. I can't believe our arm libraries don't work with the new @azure/identity library. Not your issue, but I will ping someone in central SDK about this. Frustrating.